node/node-netbox.inc.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+require_once('Horde/String.php');
+require_once('Horde/Text/Diff.php');
+require_once('Horde/Text/Diff/Engine/Native.php');
+require_once('Horde/Text/Diff/Op/Base.php');
+require_once('Horde/Text/Diff/Op/Add.php');
+require_once('Horde/Text/Diff/Op/Copy.php');
+require_once('Horde/Text/Diff/Op/Change.php');
+require_once('Horde/Text/Diff/Renderer.php');
+require_once('Horde/Text/Diff/Renderer/Inline.php');
+
+include("inc/config-call.inc.php");
+
+$NETBOX_API_URL = $CONFIG['URL']['NETBOX'] . "api";
+$NETBOX_URL = $CONFIG['URL']['NETBOX'];
+
+class pNetbox
+{
+    function header($NODE, $SHORT)
+    {
+        return("Netbox");
+    }
+
+	private function netbox_query($PATH,$ARGS)
+	{
+		global $NETBOX_API_URL;
+		
+		$url = curl_init($NETBOX_API_URL . $PATH . $ARGS);
+		curl_setopt($url, CURLOPT_CONNECTTIMEOUT,5);
+		curl_setopt($url, CURLOPT_RETURNTRANSFER,true);
+		curl_setopt($url, CURLOPT_CAPATH,"/etc/grid-security/certificates");
+		$out = curl_exec($url);
+		if (curl_error($url)) {
+			echo curl_error($url);
+			echo "curl error";
+			return null;
+		}
+		curl_close($url);
+		
+		$json_out = json_decode($out,true);
+		
+		// Do not have a count value or a netbox id - something is wrong
+		if (!isset($json_out['count']) && !(isset($json_out['id']))) { 
+			return null;
+		}
+		
+		// We have an id so just return the json object as it was a direct request
+		if (isset($json_out['id'])) {
+			return $json_out;
+		}
+		
+		if (isset($json_out['count'])) {
+			if ($json_out['count'] == 0) {
+				// Search returned no values so return null
+				return null;
+			} else {
+				// Search returned multiple values so return the array
+				return $json_out['results'];
+			}
+		}
+	}
+	
+	// Get Netbox info for a single machine
+	// To the Netbox API this is technically a search
+	// So be careful we only get one response back	
+    private function get_netbox_info($machineName)
+    {
+        global $SQL;
+
+		$query=$this->netbox_search("/dcim/devices/",array("name"=>$SQL->real_escape_string($machineName)));
```

Comment:
Does this really need to be escaped for MySQL use with `real_escape_string`?
`$NODE` is already sanitised by the node wrapper.

Reply:
That was reused from the MagDB code

Reply:
I know, but it's not being passed to a DB, so shouldn't need it.